### PR TITLE
checkOverlap have standard method and sampling method

### PR DESCRIPTION
**Pull request by krpalash**
I included both method for checkOverlaps. we can't use both method simultaneously

### CODE REVIEW

**Comment by ast0815** [comment]
Can you please re-base your changes or merge the changes on master into your branch? Right now it is in conflict.
Also, please add your change to the `CHANGELOG.md`.

**Review by ast0815 — CHANGES_REQUESTED** [review]
Can you make the indentation consistent in your changes? I.e. replace the tabs with the right number of spaces?

**Comment by ast0815** [comment]
You write that both methods do not work at the same time. That is unfortunate. Instead of having to comment out the unused method, it would be much better to be able to choose the method with a command line flag. That way we can just run the same macro twice in the automated testing and do both checks.

Also, you still need to add your changes to the changelog.

**Comment by krpalash** [comment]
how could I add changes to the changelog? what's the command for that?

**Comment by ast0815** [comment]
There is no command for it. You just need to add a line in the file `CHANGELOG.md`, describing what you changed. It is nothing special, just a text file to keep track of what changes in the software over time.

**Comment by krpalash** [comment]
I commented on changelog. let me know ,please

**Comment by ast0815** [comment]
Hi @krpalash, it seems like you changed the indentation in the whole file, usually it is better to stick with the indentation convention already present. Otherwise it is more difficult to figure out what lines actually changed logically in the future. In this case it is almost a complete rewrite of the file anyway, so I guess it does not matter much.

The indentation seems inconsistent though. Some levels are 2 spaces and some are just one. Could you make it consistent please?

Also, please address the other changes I have requested before:

- Change the line in the CHANGELOG.md to adhere to the format of that file.
- Implement a check whether we are running in batch mode for both overlap checking methods, so they can both be used in the automated tests. See how it is done for the sampling method.

**Comment by krpalash** [comment]
Regarding both method. I will try to implement. I already fixed indentation.

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,8 +20,8 @@ Geometry releases will be tagged as `Descriptive_tag_v_X.Y.Z`.
 - Fixed Overlaps in TMS geometry.
 
 ## Changed
-
 - `checkOverlaps.C` now uses sampling method.
+'checkOverlaps.C' now uses 1 to run sampling method. Other than 1 to run standard method.
```
Comment (ast0815) [inline CHANGELOG.md:24]:
```suggestion
- `checkOverlaps.C` now takes an optional argument to choose the overlap checking method. 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,8 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to a modified version of [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 Geometry releases will be tagged as `Descriptive_tag_v_X.Y.Z`.
-
+##[checkOverlaps.c]
```
Comment (ast0815) [inline CHANGELOG.md:8]:
```suggestion

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,8 @@ All notable changes to this project will be documented in this file.
...
+-Standard method and sampling method have been integrated to overlaps of geometry.
```
Comment (ast0815) [inline CHANGELOG.md:9]:
```suggestion

```

File: checkOverlaps.C [hunk 4]
```diff
@@ -1,28 +1,47 @@
...
+       cout<<"================= Overlap messages will duplicate below ================"<<endl;
+       cout<<"=================     Overlaps are in units of cm       ================"<<endl;
+       TCanvas* c = new TCanvas("c","sampling method");
+       overlap->Draw("");
+       TCanvas* cogl = new TCanvas();
+       overlap->Draw("ogl");
+       cout<<"========================       Done!       =============================\n\n\n"<<endl;
+     }
+   }
+}
```
Comment (ast0815) [inline checkOverlaps.C:27]: This curly bracket is at the wrong indentation. Also, your first indentation level is one space, while the second level is two spaces. Please make this consistent.

**Review by ast0815 — CHANGES_REQUESTED** [review]

**Pull request merged**